pkg/utils/node/finalizer.go

```diff
@@ -0,0 +1,17 @@
+package node
+
+import (
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func RemoveFinalizer(node *v1.Node, index int) {
```

Comment:
Would it make more sense to pass in the string of the finalizer you want to delete here since it's just a string, right? Passing in an index seems a little too low level since you have to loop to find the index based on a string comparison anyways. You could get rid of the for loop in the termination logic too, I think.